### DISCUSSION TITLE
Redeploy FinOps Hub

### DISCUSSION THREAD

**Discussion by equeirokyndryl** [microsoft/finops-toolkit#934] [Q&A] [answered]
A few months ago i deployed the FinOps Hub in my environment (v0.3). Now, i have to move the services to another directory, since the current directory where the Hub is hosted might be deleted soon. I tried to move the resources to the new Tenant, but it is not possible due some components like the Managed Identities and Keyvault. Based on that facts, i deployed a new hub in the new tenant, and the available version of the hub is v0.4, and not sure why, but the new hub is failing to collect the data due some issue in the pipeline. The given error is _"Operation on target ForEach Export Scope failed: The function 'length' expects its parameter to be an array or a string. The provided value is of type 'Object'._" and the pipeline is the config_ExportData. 

Can someone give a hand to solve it?

![image](https://github.com/user-attachments/assets/a727f49e-8a3d-463c-a484-9ae5c51bbc3f)

**Comment by flanakin** [comment]
@equeirokyndryl Just to confirm, did you configure managed exports for this? Assuming so, can you share what your config/settings.json looks like? Feel free to block out part of the scope IDs. I have a suspicion the array of scopes is being written as a single instance based on this error.

As a workaround, have you tried configuring exports manually? Just want to make sure you're not completely blocked while we resolve this.

**Reply by equeirokyndryl** [reply]
Hello @flanakin. So, the first Hub was created, and no additional configurations have been done on it. It was deployed as it is and started collecting the data from Azure. But the new Hub fails with the standard configs. 
(I think at this point i have to mention i have no experience with Data Factory :) )

**Reply by equeirokyndryl** [reply]
Hey, I got the point (with MS help). I just configured the Cost exports, and now seems the hub is working. I was thinking the Data Factory was responsible by collecting the data, and was expecting for it. Now i have the data into the storage account, and seems the pipeline are working properly.

**Reply by flanakin** [reply]
I have a feeling I know what the issue is. A couple other people are running into an issue with the `scopes` property in the settings.json file not being an array. Can you confirm that the `config/settings.json` file surrounds the `scopes` property with `[` and `]`?

**Reply by equeirokyndryl** [reply]
This is the `settings.json`

`{
  "$schema": "https://aka.ms/finops/hubs/settings-schema",
  "type": "HubInstance",
  "version": "0.4",
  "learnMore": "https://aka.ms/finops/hubs",
  "scopes": {
    "scope": ""
  },
  "retention": {
    "ingestion": {
      "months": 13
    },
    "msexports": {
      "days": 0
    }
  }
}`

But as i said in the last command above, the issue was the Costs Exports. I didn't create in the beginning. It was really my bad. After creating the Costs Exports, now the Hub is working properly. So far so good.

**Reply by ro100e** [reply, marked as answer]
@flanakin This was solved yesterday, I was on a session with @equeirokyndryl, we just needed to create some exports for the new tenant.
